include/boost/math/differentiation/autodiff_cpp11.hpp

```diff
@@ -230,15 +231,19 @@ template<typename RealType, size_t Order>
 template<typename RootType>
 fvar<RealType,Order>& fvar<RealType,Order>::negate_cpp11(std::true_type, const RootType&)
 {
-  std::for_each(v.begin(), v.end(), [](RealType& r) { r.negate(); });
+  for (auto& r : v) {
+    r.negate();
+  }
```

Comment:
Why this change?

Reply:
I did it for readability, since the lambda expression in this case just wrapping the body of the for-range loop (a single call to a member function with a side effect). If you would prefer the `std::for_each`, I can revert.

Reply:
Interesting reading on the topic: https://www.fluentcpp.com/2018/03/30/is-stdfor_each-obsolete/

Reply:
I’m a big fan of that site. Since readability is primary to me if no alternative holds any particular performance or efficiency advantage over the others, the loop body is sufficiently simple that achieving any level of abstraction beyond what’s there would be extremely difficult, primarily since we’re calling a member function rather than passing a container element into a function (ie negate(r) for each r in v).

I’m not dogmatic on this choice; of Matthew would prefer std::for_each, I’m fine reverting. The substance of these pull requests are merely suggestions to the main author.

Reply:
That's fine for the `_cpp11.hpp` file - please don't make this change to `autodiff.hpp`.

For what it's worth, the reason I originally started using `std::for_each` was that `cppcheck` complained about this in `operator/=(const root_type& ca)`:

    for (RealType& x : v)
        x /= ca;

Changing it to `std::for_each(v.begin(), v.end(), [&ca](RealType& x) { x /= ca; });` satisfied it. I don't recall off-hand what the complaint was.